tests/unit/TransactionalFolioNumber.spec.ts

```diff
@@ -23,42 +25,122 @@ const tests = [
       transactionalFolioNumber: '4321',
       doValidate: false
     }
+  },
+  {
+    props: {
+      accountFolioNumber: '',
+      transactionalFolioNumber: 'x'.repeat(50), // valid length
+      doValidate: true
+    },
+    shouldBeValid: true
+  },
+  {
+    props: {
+      accountFolioNumber: '',
+      transactionalFolioNumber: 'x'.repeat(51), // invalid length
+      doValidate: true
+    },
+    shouldBeValid: false
+  },
+  {
+    props: {
+      accountFolioNumber: '',
+      transactionalFolioNumber: '',
+      doValidate: true
+    },
+    newValue: 'x'.repeat(50), // valid length
+    shouldBeValid: true
+  },
+  {
+    props: {
+      accountFolioNumber: '',
+      transactionalFolioNumber: '',
+      doValidate: true
+    },
+    newValue:  'x'.repeat(51), // invalid length
+    shouldBeValid: false
   }
 ]
 
+// Conditional check for specific test scenarios
+const itIf = (condition: boolean) => condition ? it : it.skip
+
 for (const [i, test] of tests.entries()) {
   describe(`Transactional Folio Number component, test #${i}`, () => {
-    let wrapper
+    let wrapper: Wrapper<any>
 
-    beforeAll(() => {
-      wrapper = shallowWrapperFactory(TransactionalFolionNumber, test.props)
+    beforeEach(() => {
+      wrapper = wrapperFactory(TransactionalFolionNumber, test.props)
     })
 
-    afterAll(() => {
+    afterEach(() => {
       wrapper.destroy()
     })
 
     it('displays the component', () => {
       expect(wrapper.find(TransactionalFolionNumber).exists()).toBe(true)
     })
 
-    it('displays the default account folio number', () => {
-      if (!test.props.transactionalFolioNumber && test.props.accountFolioNumber) {
-        expect(wrapper.vm.localFolioNumber).toBe(test.props.accountFolioNumber)
-      }
+    itIf(!test.props.transactionalFolioNumber && !!test.props.accountFolioNumber)('displays the default account folio number', () => {
+      expect(wrapper.vm.localFolioNumber).toBe(test.props.accountFolioNumber)
+    })
+
+    itIf(!!test.props.transactionalFolioNumber)('displays the transactional folio number', () => {
+      expect(wrapper.vm.localFolioNumber).toBe(test.props.transactionalFolioNumber)
+    })
+
+    itIf(!test.props.doValidate)('is valid when validation is disabled', () => {
+      expect(wrapper.vm.isValid).toBe(true)
+      expect(wrapper.find('#transactional-folio-number-container').classes('invalid-section')).toBe(false)
+      expect(wrapper.find('label').classes('error-text')).toBe(false)
+    })
+
+    // FUTURE: fix this; for some reason, the component under test can't read `this.$refs.folioNumberInput`
```

Comment:
@cameron-freshworks Have you ever seen this??? The getter in the component always sees `undefined` for the ref. I tried it as a method as well, no change. This looks the same as other test files where we seem to do this successfully, but no luck for me today.

Reply:
hmm, i'm not sure i've seen this issue specifically but inside the jest test wouldn't the reference to the `ref` be something like `wrapper.vm.$refs.refNameDefinedOnTheComponent`?

My first thought was to also use a regular mount (which you've already done of course :) ) followed by using the defined ref on the component. Other than that i'm not sure 🤷

Reply:
The test can actually read `wrapper.vm.$refs.refNameDefinedOnTheComponent`. It's the actual code that can't (so the getter returns the wrong value).
🤷🤷🤷